infra/Pos/Communication/Relay/Types.hs

```diff
@@ -44,10 +45,17 @@ instance Buildable PropagationMsg where
     build (DataOnlyPM conts) =
         Buildable.build conts
 
--- | Queue of InvMsges which should be propagated.
-type RelayPropagationQueue = TBQueue (Microsecond, PropagationMsg)
+data RelayContext m = RelayContext
```

Comment:
This approach with `m` parameter contradicts to what we are currently using.
Currently used approach is to have:
```
class MonadRelay m where
    rlyIsPropagation :: m Bool
    rlyEnqueue :: (PropagationMsg -> Maybe NodeId -> m ())
    rlyDequeue :: (m (PropagationMsg, Maybe NodeId, NodeId, Microsecond))
```

And also we have `data RelayContext = RelayContext Bool AllOtherDataToImplementRelayMethods…`.

Which approach is better is a separate topic, probably not for this discussion. My point is that we shouldn't mix these approaches.

cc @georgeee